cpu/nrf52/periph/i2c.c

```diff
@@ -169,27 +169,44 @@ int i2c_write_regs(i2c_t dev, uint16_t addr, uint16_t reg,
 {
     assert((dev < I2C_NUMOF) && data && (len > 0) && (len < 255));
 
-    if (flags & (I2C_NOSTART | I2C_REG16 | I2C_ADDR10)) {
+    if (flags & (I2C_NOSTART | I2C_ADDR10)) {
         return -EOPNOTSUPP;
     }
+
     /* the nrf52's TWI device does not support to do two consecutive transfers
      * without a repeated start condition in between. So we have to put all data
      * to be transferred into a temporary buffer
      *
      * CAUTION: this might become critical when transferring large blocks of
      *          data as the temporary buffer is allocated on the stack... */
-    uint8_t buf_tmp[len + 1];
```

Comment:
urgh so this was incidentally putting up to 256 bytes of memory on the stack and using VLA on top?
I know this was not introduced by you, but to avoid nasty surprises I would make this a 

```C
    static uint8_t tx_buf[256];
```

Reply:
Yeah the VLA it is not the best solution in my opinion either... 

The fixed size buffer at the top seems more reasonable. There's already a size check on the data length parameter so I don't see any reason why we couldn't do this instead.